Add codemod ID (#35)
pixee/codemodder-specs: codetf.md

Note that like SARIF, this format is not intended to be a replacement for a diagnostic log. It's not intended to have anything more than minimum diagnostics to help with reproducibility.

# Structure

It may help to understand the major components of CodeTF from a high levels first before exploring or attemptin to implement the specification. The `results` and `changeset` fields can be seen as a series of patches against a project's directory. Each patch builds on any previous patches seen. Therefore, applying a patch from the middle of a `changeset` without the others may be invalid. Multiple locations can be changed in a single file within the scope of a single codemod and be represented by a single `changeset` array entry.

# Codemod IDs

> Codemods are uniquely identified by an ID, which is represented in CodeTF as the `codemod` property of the `result` object.

Reviewer: Can we go as far as to call the ID a "URI", since these identifiers do conform to that structure?

```suggestion
Codemods are uniquely identified by a URI, which is represented in CodeTF as the `codemod` property of the `result` object.
```

IDs are descriptive and must conform to the following schema: `<origin>:<language>/<name>`

Each component of the ID has a particular meaning:

> * `<origin>`: Origin describes the source of the analysis or transformation. For example, "find and fix" codemods provided by Pixee are labelled with the origin "pixee". Codemods that remediate issues found by a static analysis tool might be labelled with the origin corresponding to that tool name (e.g. "semgrep" or "codeql"). Implementers of custom codemods may use a unique identifier that is specific to their organization or tool.

Reviewer 2: I don't like `scanner` because there's no reason codemodder couldn't take input from non-scanner sources of code information, like observability tools, IAST, etc.

> * `<origin>`: Origin describes the source of the analysis or transformation. For example, "find and fix" codemods provided by Pixee are labelled with the origin "pixee". Codemods that remediate issues found by a static analysis tool might be labelled with the origin corresponding to that tool name (e.g. "semgrep" or "codeql"). Implementers of custom codemods may use a unique identifier that is specific to their organization or tool.

Reviewer: `detector`?

> * `<origin>`: Origin describes the source of the analysis or transformation. For example, "find and fix" codemods provided by Pixee are labelled with the origin "pixee". Codemods that remediate issues found by a static analysis tool might be labelled with the origin corresponding to that tool name (e.g. "semgrep" or "codeql"). Implementers of custom codemods may use a unique identifier that is specific to their organization or tool.

Author: I like `detector`. This fits with the way we talk about it in codemodder architecture.

> * `<origin>`: Origin describes the source of the analysis or transformation. For example, "find and fix" codemods provided by Pixee are labelled with the origin "pixee". Codemods that remediate issues found by a static analysis tool might be labelled with the origin corresponding to that tool name (e.g. "semgrep" or "codeql"). Implementers of custom codemods may use a unique identifier that is specific to their organization or tool.

Reviewer: I love it when we use the same words everywhere 😊
